File: Queue/Queue.cpp
#include <iostream>
#include "Queue.h"  // Assuming your Queue class is defined in Queue.h

int main() {
    Queue<int> q(10);  // Create a queue for integers

    std::cout << "Enqueuing elements{1,2,3,4}..." << std::endl;
    q.enqueue(1);
    q.enqueue(2);
    q.enqueue(3);

    std::cout << "Current front element: " << q.peek() << std::endl;  // Should print 1

    std::cout << "Dequeuing remaining elements..." << std::endl;
    q.dequeue();
    std::cout << "New front element after one dequeue: " << q.peek() << std::endl;  // Should print 2

    std::cout << "Checking if queue is empty: " << (q.isEmpty() ? "Yes" : "No") << std::endl;  // Should print No

    std::cout << "Dequeuing remaining elements..." << std::endl;
    q.dequeue();
    q.dequeue();  // Queue should be empty after this

    std::cout << "Checking if queue is empty after dequeuing all elements: " << (q.isEmpty() ? "Yes" : "No") << std::endl;  // Should print Yes

    try {
        std::cout << "Attempting to peek into an empty queue: ";
        std::cout << q.peek() << std::endl;
    } catch (const std::runtime_error& e) {
        std::cout << e.what() << std::endl;  // Should catch and print "Queue is empty"
    }

    return 0;
}

File: Queue/Python/Queue.py
class Queue:
    def __init__(self, capacity=None):
        self.capacity = capacity
        self.q = []

    def enqueue(self, value):
        if self.capacity is not None and len(self.q) >= self.capacity:
            raise RuntimeError("Queue is full")
        self.q.append(value)

    def dequeue(self):
        if not self.isEmpty():
            return self.q.pop(0)
        return "Queue is empty"

    def peek(self):
        if self.isEmpty():
            return "Queue is empty"
        return self.q[0]
    
    def isEmpty(self):
        return len(self.q) == 0

    def __len__(self):
        return len(self.q)


def main():
    q = Queue()  # Create a queue for integers
    print("Enqueuing elements{1,2,3,4}...")
    q.enqueue(1)
    q.enqueue(2)
    q.enqueue(3)
    q.enqueue(4)  # This was not included in your original C++ enqueuing, but added as per the printed message

    print(f"Current front element: {q.peek()}")  # Should print 1

    print("Dequeuing...")
    q.dequeue()
    print(f"New front element after one dequeue: {q.peek()}")  # Should print 2

    print(f"Checking if queue is empty: {'Yes' if q.isEmpty() else 'No'}")  # Should print No

    print("Dequeuing remaining elements...")
    q.dequeue()
    q.dequeue()
    q.dequeue()  # Added one more dequeue to empty the queue after adding 4th element

    print(f"Checking if queue is empty after dequeuing all elements: {'Yes' if q.isEmpty() else 'No'}")  # Should print Yes

    try:
        print("Attempting to peek into an empty queue: ", end="")
        print(q.peek())
    except IndexError as e:
        print(e)  # Should catch and print "Queue is empty"

if __name__ == "__main__":
    main()